### DISCUSSION TITLE
Add the ability to fetch a user's submissions

### DISCUSSION THREAD

**Discussion by ghost** [thislooksfun/snoots#22] [Q&A] [answered]
A project I'm currently working on requires fetching the list of posts created by a given user. I would love to see this functionality implemented.

I believe the API endpoint is https://www.reddit.com/dev/api/#GET_user_{username}_submitted

Cheers.

**Comment by thislooksfun** [comment, marked as answer]
It's definitely on the list! I'll try to bump it towards the front. Can't make any promises, but it should hopefully be out in a few days.

**Reply by thislooksfun** [reply]
@messier102 [v1.0.0-dev.11](https://github.com/thislooksfun/snoots/releases/tag/v1.0.0-dev.11) just added [Client.users.getPosts()](http://thislooks.fun/snoots/docs/latest/classes/usercontrols#getposts)!

I realized that I forgot to give it a default sort, but that's been fixed in [v1.0.0-dev.12](https://github.com/thislooksfun/snoots/releases/tag/v1.0.0-dev.12).

Thanks for using snoots, and let me know ~~when~~ if you come across more missing features!

**Reply by ghost** [reply]
Thank you very much for the quick release! Excited to build more features with snoots.

**Reply by ghost** [reply]
I am running into a bit of an issue with calling `getPosts()` on a user who doesn't have any submissions. As far as I understand, it should return an empty `Listing`, but I instead get a non-empty listing with a single `undefined` item, which makes my `for await` loop explode. Could you look into that, please?

**Reply by thislooksfun** [reply]
Yeah that's not right. Can you give me the username so I can test it?

**Reply by ghost** [reply]
You can try `_messier102`, but it seems to be happening with any account with no posts.

**Reply by thislooksfun** [reply]
Fixed in .13! (give it a minute to release)

**Reply by ghost** [reply]
Can confirm, working correctly now 😄 Thank you kindly.
